ci_framework/roles/hive/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Fails if the oc client is lesser than 4
+
+- name: Ensure hive output directory exists
+  ansible.builtin.file:
+    path: "{{ cifmw_hive_artifacts_dir }}"
+    state: directory
+
+- name: Performing {{ cifmw_hive_platform }} {{ cifmw_hive_action }}
+  ansible.builtin.include_tasks: "{{ cifmw_hive_platform}}_{{ cifmw_hive_action }}.yml"
+  when:
+    - cifmw_hive_platform is defined
```

Comment:
is defined
cifmw_hive_platform in ["openstack", "baremetal"] 

or you could add a separate task to enforce that like failed_when: cifmw_hive_platform not in

Reply:
ack, adding some asserts for the vars that are stated as required in the readme